controllers/cluster_controller.go

```diff
@@ -211,8 +219,43 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 	}
 
 	defer func() {
+		conditionResult, err := r.reconcileClusterCondtions(ctx, log, cluster)
+		if err != nil {
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+
+		// Use the condition result if the last reconcile does not specify requeue or an interruption
+		if result.IsZero() && reterr == nil {
+			result = conditionResult
+		}
+
+		// Always update the readyCondition by summarizing the state of other conditions.
+		conditions.SetSummary(cluster,
+			conditions.WithConditions(
+				clusterv1.ControlPlaneReadyCondition,
```

Comment:
is there a reason to look for capi ready here instead of eksa ready condition for eksa cluster?

Reply:
it looks for the a condition with condition.Type == "ControlPlaneReady". The package does not matter here since the strings are the same. We are going to use our own types for conditions  instead for though